framework/ui/src/main/java/com/google/android/catalog/framework/ui/CatalogScreen.kt

```diff
@@ -104,37 +118,91 @@ internal fun CatalogScreen(
         }
     }
 
+    val isExpandedScreen = isExpandedScreen()
+    val saver = listSaver<CatalogSample?, String>(
+        save = { listOf(it?.route.orEmpty()) },
+        restore = { list -> displayedSamples.find { it.route == list.firstOrNull() } }
+    )
+    var selectedSample by rememberSaveable(isExpandedScreen, stateSaver = saver) {
+        mutableStateOf(null)
+    }
+
     Scaffold(
         topBar = {
-            if (searchState) {
-                SearchTopAppBar(
-                    searchTerm = searchTerm,
-                    focusRequester = focusRequester,
-                    onSearch = {
-                        searchState = false
-                    },
-                    onClear = {
-                        searchState = false
-                    },
-                    onValueChange = {
-                        searchTerm = it
+            AnimatedContent(
+                targetState = searchState,
+                transitionSpec = {
+                    if (targetState) {
+                        expandHorizontally { it / 8 } + fadeIn(tween(100)) with fadeOut()
+                    } else {
+                        fadeIn(tween(500, delayMillis = 100)) with
+                            shrinkHorizontally(tween(400)) { it / 8 } + fadeOut(tween(400))
                     }
-                )
-            } else {
-                CatalogTopAppBar(onSearch = { searchState = true })
+                },
+                contentAlignment = Alignment.CenterEnd
+            ) {
+                if (it) {
+                    SearchTopAppBar(
+                        searchTerm = searchTerm,
+                        focusRequester = focusRequester,
+                        onSearch = {
+                            searchState = false
+                        },
+                        onClear = {
+                            searchState = false
+                        },
+                        onValueChange = {
+                            searchTerm = it
+                        }
+                    )
+                } else {
+                    CatalogTopAppBar(
+                        isDualPane = isExpandedScreen,
+                        selectedSample = selectedSample,
+                        onSearch = { searchState = true },
+                        onExpand = { launchSample(selectedSample!!) }
+                    )
+                }
             }
         }
     ) { paddingValues ->
-        SamplesList(
-            paddingValues,
-            filters,
-            selectedFilters,
-            displayedSamples,
-            catalogSettings.cardAppearance
-        ) {
-            launchSample(it)
-            searchState = false
-        }
+        AdaptivePane(
+            paddingValues = paddingValues,
+            start = {
+                SamplesList(
+                    paddingValues,
```

Comment:
It seems slightly suspicious here to pass the `paddingValues` to both the `SamplesList` and the `AdaptivePane`. It feels like it should only be handled by one of them?

Reply:
Good point. I will apply it only to the adaptive pane